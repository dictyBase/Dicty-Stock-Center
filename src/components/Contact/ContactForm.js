// @flow
import React from "react"

/**
 * DSC contact form
 */

const ContactForm = () => (
  <center>
    <p>
      For questions, comments, or suggestions, please send us an email&nbsp;
      <a
        href="mailto:dictystocks@northwestern.edu?Subject=Question"
        target="_top">
        (dictystocks@northwestern.edu)
      </a>
      .
    </p>
    <p>We will be adding a working contact form soon.</p>
  </center>
)

export default ContactForm
