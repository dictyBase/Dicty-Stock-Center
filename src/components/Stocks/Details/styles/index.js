import { makeStyles } from "@material-ui/core/styles"
import { green } from "@material-ui/core/colors"

const useStyles = makeStyles(({ palette, spacing }) => ({
  root: {
    width: "90%",
    margin: "auto",
  },
  header: {
    textAlign: "center",
  },
  backButton: {
    marginTop: "10px",
    textAlign: "left",
    color: "#5f6368",
    "&:hover": {
      backgroundColor: "transparent",
      color: "inherit",
    },
  },
  cardHeader: {
    color: "#002f5e",
  },
  details: {
    "&:nth-child(odd)": {
      backgroundColor: "#F4F6F8",
    },
    "@media (max-width: 420px)": {
      display: "block !important",
    },
  },
  listItem: {
    "@media (max-width: 420px)": {
      marginBottom: "10px",
    },
  },
  list: {
    width: "100%",
    padding: "0px !important",
  },
  availabilityCard: {
    paddingTop: "5px",
    textAlign: "center",
  },
  cartBtnRow: {
    padding: "16px",
  },
  unavailableText: {
    padding: "16px",
    textAlign: "justify",
  },
  moreStrainsCard: {
    marginTop: "16px",
    padding: "5px",
    textAlign: "center",
  },
  options: {
    paddingBottom: "8px",
    display: "flex",
    justifyContent: "center",
  },
  secondaryText: {
    paddingTop: "8px",
  },
  link: {
    textDecoration: "none",
    color: palette.primary.main,
  },
  checkoutBtn: {
    borderStyle: "none !important",
    borderRadius: "0px !important",
    fontSize: "0.9rem !important",
  },
  arrowIcon: {
    marginRight: "8px",
  },
  id: {
    display: "flex",
    justifyContent: "center",
  },
  snackbar: {
    width: "300px",
    backgroundColor: green[600],
    fontSize: "0.8rem",
    display: "flex",
    alignItems: "center",
    paddingTop: "0px",
    paddingBottom: "0px",
  },
  checkIcon: {
    fontSize: "0.8rem",
    marginRight: spacing(1),
  },
  closeIcon: {
    fontSize: "0.9rem",
  },
  unavailableIcon: {
    marginRight: spacing(2),
  },
  copyIcon: {
    fontSize: "0.8rem",
    "&:hover": {
      backgroundColor: "transparent",
      color: "rgba(0, 0, 0, 0.87)",
    },
  },
  detailsItem: {
    fontWeight: 420,
  },
  relatedItem: {
    display: "flex",
    justifyContent: "center !important",
  },
  addToCartBtn: {
    backgroundColor: green[600],
    color: "#fff",
    paddingLeft: "16px",
    paddingRight: "16px",
    "&:hover": {
      backgroundColor: green[700],
      color: "#fff",
    },
  },
  removeBtn: {
    marginTop: "5px",
  },
  quantity: {
    display: "flex",
    justifyContent: "space-evenly",
    alignItems: "center",
    padding: "16px",
  },
  maxItems: {
    color: palette.secondary.main,
  },
}))

export default useStyles
