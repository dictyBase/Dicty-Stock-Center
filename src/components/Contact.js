// @flow
import React from "react"
import { Helmet } from "react-helmet"
import { withStyles } from "@material-ui/core/styles"
import Grid from "@material-ui/core/Grid"

const styles = theme => ({
  header: {
    marginTop: "25px",
    marginBottom: "25px",
    borderBottom: "1px solid #eee",
    textAlign: "center",
  },
  container: {
    paddingRight: "15px",
    paddingLeft: "15px",
    marginRight: "auto",
    marginLeft: "auto",
    width: "75%",
    "@media (min-width: 1300px)": {
      width: "1270px",
    },
  },
})

type Props = {
  /** Material-UI styling */
  classes: Object,
}

/**
 * DSC contact form
 */

const Contact = (props: Props) => {
  const { classes } = props

  return (
    <>
      <Helmet>
        <title>Contact Us - Dicty Stock Center</title>
        <meta
          name="description"
          content="Contact page for Dicty Stock Center"
        />
      </Helmet>
      <Grid container className={classes.container} justify="center">
        <Grid item xs={12}>
          <div className={classes.header}>
            <h1>Contact Us</h1>
          </div>
        </Grid>
        <Grid item xs={12}>
          <center>
            <p>
              For questions, comments, or suggestions, please send us an
              email&nbsp;
              <a
                href="mailto:dictystocks@northwestern.edu?Subject=Question"
                target="_top">
                (dictystocks@northwestern.edu)
              </a>
              .
            </p>
            <p>We will be adding a working contact form soon.</p>
          </center>
        </Grid>
      </Grid>
    </>
  )
}

export default withStyles(styles)(Contact)
