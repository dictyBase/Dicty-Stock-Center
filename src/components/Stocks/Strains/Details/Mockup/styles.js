import { makeStyles } from "@material-ui/core/styles"

const useStyles = makeStyles({
  root: {
    width: "80%",
    margin: "auto",
  },
  header: {
    textAlign: "center",
  },
  leftCard: {},
  details: {
    "&:hover": {
      boxShadow:
        "inset 1px 0 0 #dadce0,inset -1px 0 0 #dadce0,0 1px 2px 0 rgba(60,64,67,.3),0 1px 3px 1px rgba(60,64,67,.15)",
      zIndex: 1,
    },
    "&:nth-child(odd)": {
      backgroundColor: "#eeeeee",
    },
  },
  listTitle: {},
  listContent: {},
  rightCard: {
    padding: "5px",
    textAlign: "center",
  },
  quantity: {
    display: "inline-flex",
    justifyContent: "center",
  },
  textField: {},
})

export default useStyles
