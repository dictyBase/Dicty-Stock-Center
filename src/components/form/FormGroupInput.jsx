// @flow
import React, { Component } from 'react'
import { Flex, Box } from 'rebass'
import { ControlLabel, FormGroup, FormControl, HelpBlock } from 'styles'

type Props = {
  field: Object,
  inputType: string,
  placeholder: string
}

export default class FormGroupInput extends Component<Props> {
  displayName = 'form group with input'

  static defaultProps = {
      inputType: 'text'
  }

  render() {
      const { field, inputType, placeholder } = this.props
      const hasError = field.touched && field.error

      return (
      <FormGroup>
        <Flex wrap justify="center">
          <Box w={ [1, 1 / 4, 1 / 4] } ml={ 1 } mb={ 2 } mr={ 1 }>
            <ControlLabel>{ this.props.children }</ControlLabel>
          </Box>
          <Box w={ [1, '60%', '60%'] } mr={ 1 } ml={ 1 }>
            <FormControl
              type={ inputType }
              {...field}
              placeholder={ placeholder }
            />
            { hasError && <HelpBlock>{ field.error }</HelpBlock> }
          </Box>
        </Flex>
      </FormGroup>
    )
  }
}
