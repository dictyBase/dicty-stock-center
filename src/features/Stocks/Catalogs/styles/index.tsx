import { makeStyles, Theme } from "@material-ui/core/styles"
import { blue } from "@material-ui/core/colors"

const useStyles = makeStyles((theme: Theme) => ({
  appBar: {
    backgroundColor: blue[800],
  },
  catalogPaper: {
    height: 600,
    width: "100%",
  },
  listHeaders: {
    position: "sticky",
    top: 0,
  },
  row: {
    borderBottom: "1px solid rgba(224, 224, 224, 1)",
    "&:hover": {
      backgroundColor: "#eeeeee",
      boxShadow:
        "inset 1px 0 0 #dadce0,inset -1px 0 0 #dadce0,0 1px 2px 0 rgba(60,64,67,.3),0 1px 3px 1px rgba(60,64,67,.15)",
      zIndex: 1,
    },
    "@media (max-width: 1024px)": {
      "& p": {
        fontSize: "0.75rem !important",
      },
    },
  },
  item: {
    paddingRight: theme.spacing(1),
  },
  trashBtn: {
    color: theme.palette.error.main,
  },
}))

export default useStyles
