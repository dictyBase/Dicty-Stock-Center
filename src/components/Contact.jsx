// @flow
import React, { Component } from "react"
import {
  PanelGroup,
  Panel,
  PanelHeader,
  PanelTitle,
  PanelBody
} from "dicty-components-panel"
import FormGroupInput from "./form/FormGroupInput"
import Comments from "./form/Comments"
import SubmitButton from "./form/SubmitButton"
import { submitEmail } from "actions/contact"
import { reduxForm } from "redux-form"
import { syncValidate } from "forms/validate/contact-form"
import { Flex, Box } from "rebass"
import FontAwesome from "react-fontawesome"
import {
  DictyHeader,
  RequiredText,
  Container,
  HorizontalForm,
  ResetButton
} from "styles"

export const fields = ["name", "email", "subject", "message"]

const theme = {
  headerBackgroundColor: "#f9f9f9",
  headerTextColor: "#000000"
}

type Props = {
  /** the fields of the contact form */
  fields: Object,
  /** function to handle submit */
  handleSubmit: Function,
  /** checks if form is submitting */
  submitting: boolean,
  /** erases the current form content */
  resetForm: Function
}

/**
 * DSC contact form
 */

class Contact extends Component<Props> {
  render() {
    const {
      fields: { name, email, subject, message },
      handleSubmit,
      resetForm,
      submitting
    } = this.props
    return (
      <Container>
        <Flex wrap justify="center">
          <Box>
            <DictyHeader>
              <h1>Contact Us</h1>
            </DictyHeader>
          </Box>
          <Box w={["90%", 3 / 4]}>
            <p>
              For questions, comments, or suggestions, please fill out the form
              below to send us an email&nbsp;
              <a
                href="mailto:dictybase@northwestern.edu?Subject=Question"
                target="_top">
                (dictybase@northwestern.edu)
              </a>
            </p>
          </Box>
          <Box w={["90%", 3 / 4]}>
            <PanelGroup theme={theme}>
              <Panel collapse>
                <PanelHeader style={{ padding: "20px" }}>
                  <PanelTitle>
                    <FontAwesome name="envelope-o" /> Email dictyBase
                  </PanelTitle>
                </PanelHeader>
                <PanelBody style={{ height: "auto" }}>
                  <HorizontalForm onSubmit={handleSubmit}>
                    <FormGroupInput field={name}>
                      <RequiredText title="required field">* </RequiredText>
                      Name:
                    </FormGroupInput>
                    <FormGroupInput field={email}>
                      <RequiredText title="required field">* </RequiredText>
                      Email:
                    </FormGroupInput>
                    <FormGroupInput field={subject}>Subject:</FormGroupInput>
                    <Comments
                      comments={message}
                      rows={"5"}
                      placeholder={"Please enter your message here"}>
                      Message:
                    </Comments>
                    <Flex>
                      <Box w={1 / 2} mr={2} mt={1}>
                        <ResetButton
                          type="button"
                          disabled={submitting}
                          onClick={resetForm}>
                          Reset
                        </ResetButton>
                      </Box>
                      <Box w={1 / 2} mt={1}>
                        <SubmitButton
                          name={"Submit "}
                          submitting={submitting}
                          icon={"paper-plane-o"}
                        />
                      </Box>
                    </Flex>
                  </HorizontalForm>
                </PanelBody>
              </Panel>
            </PanelGroup>
          </Box>
        </Flex>
      </Container>
    )
  }
}

const mapStateToProps = state => {
  const { user } = state.auth
  return {
    initialValues: {
      email: user ? user.email : ""
    }
  }
}

export default reduxForm(
  {
    form: "contact",
    fields,
    onSubmit: submitEmail,
    validate: syncValidate
  },
  mapStateToProps
)(Contact)
