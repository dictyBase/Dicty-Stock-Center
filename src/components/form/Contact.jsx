// @flow
import React from "react"
import FormGroupInput from "./FormGroupInput"
import { RequiredText } from "styles"

type Props = {
  phone: Object
}

const Contact = (props: Props) => {
  const { phone } = props
  return (
    <div>
      <FormGroupInput field={phone}>
        <RequiredText title="required field">* </RequiredText>
        Phone:
      </FormGroupInput>
    </div>
  )
}

export default Contact
