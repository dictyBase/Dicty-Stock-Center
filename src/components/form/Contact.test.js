import React from "react"
import renderer from "react-test-renderer"
import "jest-styled-components"
import Contact from "./Contact"
import FormGroupInput from "./FormGroupInput"
import { shallow } from "enzyme"
import { RequiredText } from "styles"

describe("form/Contact", () => {
  const phone = { value: "", touched: true }
  const wrapper = shallow(<Contact phone={phone} />)

  it("should render an input field for phone number", () => {
    expect(
      wrapper.contains(
        <FormGroupInput field={phone}>
          <RequiredText title="required field">* </RequiredText>Phone:
        </FormGroupInput>
      )
    ).toEqual(true)
  })
})

test("matching a snapshot of Contact", () => {
  const phone = {
    value: "555-5555",
    touched: true
  }

  const component = renderer.create(<Contact phone={phone} />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
