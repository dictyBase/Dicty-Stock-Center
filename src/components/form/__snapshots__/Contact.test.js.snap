// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of Contact 1`] = `
<div>
  <div
    className="sc-hmXxxW ggtKDs"
  >
    <div
      className="sc-bxivhb bUZGRr sc-htpNat eDtRyC sc-bdVaJa iHZvIS"
      is={null}
    >
      <div
        className="sc-ifAKCX cFlEyZ sc-bdVaJa iplmxF"
        is={null}
      >
        <label
          className="sc-ipXKqB hIQKmZ"
        >
          Phone:
        </label>
      </div>
      <div
        className="sc-ifAKCX cFlEyZ sc-bdVaJa cvzxNy"
        is={null}
      >
        <input
          className="sc-cqCuEk jbXqms"
          placeholder={undefined}
          type="text"
          value="555-5555"
        />
      </div>
    </div>
  </div>
</div>
`;
